### PR TITLE
dac: switch over to new comms API, add dac command line tool in setup

### PR DESCRIPTION
**Pull request by GravesJake**

### CODE REVIEW

File: host/greatfet/commands/greatfet_dac.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def main():
 
     # 0 - 1024
     dac_value = 1024
```
Comment (dominicgs) [inline host/greatfet/commands/greatfet_dac.py]: This value is hard coded but should be taken as an argument

File: firmware/greatfet_usb/classes/dac.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+/**
+ * Verbs for the firmware API.
+ */
+static struct comms_verb _verbs[] = {
+		{ .name = "set", .handler = dac_verb_set,
+            .in_signature = "<I", .out_signature = "",
+            .in_param_names = "value",
+            .doc = "Sets the DAC" },
```
Comment (dominicgs) [inline firmware/greatfet_usb/classes/dac.c]: Perhaps give a little more detail, e.g. "Sets the DAC value on GPIO2[3] (0-1024)"

File: firmware/greatfet_usb/classes/dac.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#include <dac.h>
+#include <pins.h>
+
+#define CLASS_NUMBER_SELF (0x105)
+
+int dac_verb_set(struct command_transaction *trans)
+{
+	// set J2_P5 up as a DAC pin
+	scu_pinmux(SCU_PINMUX_GPIO2_3, SCU_GPIO_FAST | SCU_GPIO_PUP | SCU_CONF_FUNCTION0);
+	static struct gpio_t dac_pin = GPIO(2, 3);
```
Comment (dominicgs) [inline firmware/greatfet_usb/classes/dac.c]: Should we be checking this pin out from the pin manager? (@ktemkin)

File: host/greatfet/commands/greatfet_dac.py (hunk 1 above)
Comment (mossmann) [reply host/greatfet/commands/greatfet_dac.py]: see #134

File: firmware/greatfet_usb/classes/dac.c (hunk 3 above)
Comment (GravesJake) [reply firmware/greatfet_usb/classes/dac.c]: Defaulting to ADC0_0 eliminates the need to use the pin manager

**Pull request merged**